bpython/formatter.py

```diff
@@ -106,15 +113,21 @@ def __init__(self, color_scheme, **options):
                 self.f_strings[k] += "I"
         super().__init__(**options)
 
-    def format(self, tokensource, outfile):
-        o = ""
+    def format(
+        self,
+        tokensource: Iterable[MutableMapping[_TokenType, str]],
+        outfile: TextIO,
+    ) -> None:
+        o: str = ""
         for token, text in tokensource:
+            t: Optional[_TokenType] = token
             if text == "\n":
                 continue
 
-            while token not in self.f_strings:
-                token = token.parent
-            o += f"{self.f_strings[token]}\x03{text}\x04"
+            while t not in self.f_strings:
+                if t is not None:
+                    t = t.parent
+            o += f"{self.f_strings[t]}\x03{text}\x04"
```

Comment:
`self.f_strings[None]` would raise an exception. It would be safe to bail out if `t.parent` is `None`.

Reply:
Ah, yes.  I'll take another shot.